DicomRtImportExport/Logic/vtkSlicerDicomRtReader.cxx

```diff
@@ -558,6 +580,62 @@ void vtkSlicerDicomRtReader::vtkInternal::LoadRTPlan(DcmDataset* dataset)
       currentBeamSequenceItem.getSourceAxisDistance(sourceAxisDistance);
       beamEntry.SourceAxisDistance = sourceAxisDistance;
 
+      // Get multi leaf collimators parameters: distance, number of leaf pairs, leaf position boundaries
+      DRTBeamLimitingDeviceSequenceInRTBeamsModule& rtBeamLimitingDeviceSequence = 
+        currentBeamSequenceItem.getBeamLimitingDeviceSequence();
+      if (rtBeamLimitingDeviceSequence.gotoFirstItem().good())
+      {
+        do
+        {
+          DRTBeamLimitingDeviceSequenceInRTBeamsModule::Item &collimatorItem =
+            rtBeamLimitingDeviceSequence.getCurrentItem();
+          if (collimatorItem.isValid())
+          {
+            OFString deviceType("");
+            Sint32 nofPairs = -1;
+
+            collimatorItem.getRTBeamLimitingDeviceType(deviceType);
+            if (deviceType == "MLCX" || deviceType == "MLCY")
+            {
+              Float64 sourceToMLCDistance = -1.;
+              collimatorItem.getSourceToBeamLimitingDeviceDistance(sourceToMLCDistance);
+              
+              OFCondition getNumberOfPairsCondition = collimatorItem.getNumberOfLeafJawPairs(nofPairs);
+              if (getNumberOfPairsCondition.good())
+              {
+                OFVector<vtkTypeFloat64> leafPositionBoundaries;
+                OFCondition getBoundariesCondition = collimatorItem.getLeafPositionBoundaries(leafPositionBoundaries);
+                if (getBoundariesCondition.good())
+                {
+                  double& distanceX = MultiLeafCollimator[0].Distance;
+                  double& distanceY = MultiLeafCollimator[1].Distance;
+                  unsigned int& pairsX = MultiLeafCollimator[0].NumberOfLeafJawPairs;
+                  unsigned int& pairsY = MultiLeafCollimator[1].NumberOfLeafJawPairs;
+                  std::vector<double>& boundX = MultiLeafCollimator[0].LeafPositionBoundary;
+                  std::vector<double>& boundY = MultiLeafCollimator[1].LeafPositionBoundary;
+                  if (deviceType == "MLCX")
+                  {
+                    distanceX = sourceToMLCDistance;
+                    pairsX = nofPairs;
+                    boundX.resize(leafPositionBoundaries.size());

```

Comment:
As far as I understand this is where you fill the Internal->MultiLeafCollimator[n] BeamLimitingDeviceEntries. But you have only 2 arrays, and this code here is in a loop. Is it always the same, or how it is not a problem that it gets overwritten at every subsequent run of the loop?

Reply:
I will be a problem if RTPlan has more than one BeamSequence. I assume, for the first time, it is easier to test everything with only one BeamSequence.

Reply:
You are right, it doesn't look good. I will rework that part.